framework/src/ble_manager/ble_manager_lwnl.c

```diff
@@ -0,0 +1,391 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <errno.h>
+#include <debug.h>
+#include <net/if.h>
+#include <tinyara/lwnl/lwnl.h>
+#include <tinyara/ble/ble_manager.h>
+#include <tinyara/net/if/ble.h>
+#include "ble_manager_log.h"
+
+#define BLE_INTF_NAME "ble0"
+
+static inline int _send_msg(lwnl_msg *msg)
+{
+	int fd = socket(AF_LWNL, SOCK_RAW, LWNL_ROUTE);
+	if (fd < 0) {
+		BLE_LOG_ERROR("send lwnl msg open error(%d)\n", get_errno(errno));
+		return -1;
```

Comment:
We define many kind of errno, so you should check errno first, and then return error type which you defined. below is an example
```
int err = errno;
if (err == EBUSY) {
    return BLE_DEVICE_BUSY;
}
```

Reply:
"_send_msg" is the internal function and if it returns negatetd value, the caller will return "TRBLE_FILE_ERROR".